### DISCUSSION TITLE
Are user id, project id and organization id globally unique?

### DISCUSSION THREAD

**Discussion by gsoec** [zitadel/zitadel#5997] [Q&A] [answered]
Could anyone please point out where I can find the code related to the ID generator? 
Thank you.

**Comment by fforootd** [comment, marked as answer]
All IDs generated by ZITADEL should be unique and collision resistant (at least to a high probability 😁 ).
We use a concept originally called Twitter Snowflake which has a Go implementation called [Sonyflake](https://github.com/sony/sonyflake).

You should find the implementation here https://github.com/zitadel/zitadel/blob/main/internal/id/sonyflake.go

**Reply by gsoec** [reply]
Thanks @fforootd
